x/pool-incentives/keeper/keeper.go

```diff
@@ -120,10 +120,16 @@ func (k Keeper) CreateConcentratedLiquidityPoolGauge(ctx sdk.Context, poolId uin
 		return err
 	}
 
+	// Although the pool id <> gauge "NoLock" link is creted in CreateGauge,
+	// we create an additional "ByDuration" link here for tracking
+	// internal incentive "NoLock" gauges
+	incentivesEpochDuration := k.incentivesKeeper.GetEpochInfo(ctx).Duration
+	k.SetPoolGaugeIdByDuration(ctx, poolId, incentivesEpochDuration, gaugeId)
```

Comment:
Can you explain a little more why we need two separate gauges for concentrated pools?

Reply:
These are not gauges.

These are pool id <> gauge id links

We need one for general purposes because "NoLock" gauges are by design associated with a pool.

We also need one for internal gauges to plug into the existing model. The current model assumes that an internal gauge has a pool id <> gauge id link created by this function: `k.SetPoolGaugeIdByDuration`